sesman/sessionrecord.c

```diff
@@ -49,31 +50,56 @@ add_xtmp_entry(int pid, const char *line, const char *user, const char *rhostnam
 {
     _utmp ut;
     struct timeval tv;
-
-    memset(&ut, 0, sizeof(ut));
+    char *hostname = 0;
+
+    /* The string rhostname containt too much data, only get the ip
+     * the format is
+     * "2001:123:12:1234:1234:1234:1234:1234:53194 - socket: 12"
+     * "::ffff:99.99.9.999:51165 - socket: 12"
+     * "99.99.9.999:51165 - socket: 12"
+     *
+     * So the IP is the string up the two last colons
+     */
```

Comment:
Communication between xrdp and sesman should be changed. I and speidy planning to do that. This part will be unneeded in future.